buchschloss/gui2/main.py

```diff
@@ -24,62 +26,34 @@
 from . import forms
 from . import widgets
 from . import actions
-from .actions import generic_formbased_action, ShowInfoNS, show_BSE
+from . import common
+from .actions import generic_formbased_action, ShowInfo
 
 
-class NSWithLogin:
-    """Wrap around an ActionNamespace providing the current login"""
-    def __init__(self, ans: T.Type[core.ActionNamespace]):
-        self.ans = ans
-
-    def __getattr__(self, item):
-        val = getattr(self.ans, item)
-        if callable(val):
-            return lambda *a, **kw: val(*a, login_context=app.current_login, **kw)
-        else:
-            return val
-
-
-class ActionTree:
-    def __getattr__(self, name):
-        if name not in self.subactions:
-            setattr(self, name, None)
-        return self.subactions[name]
-
-    def __setattr__(self, name, value):
-        if isinstance(value, __class__):  # noqa
-            self.subactions[name] = value
-        else:
-            self.subactions[name] = type(self)(value)
-
-    def __init__(self, action=None):
-        super().__setattr__('subactions', {})
-        super().__setattr__('action', action)
-        super().__setattr__('__name__', repr(self))
-
-    def __call__(self, *args, **kwargs):
-        def call_wrapper_for_unknown_reasons(func):
-            # this also works with a class defining __call__,
-            # it might be a problem with ?? the NAME ??
-            return lambda *a, **kw: func(*a, **kw)
-
-        CWFUR = call_wrapper_for_unknown_reasons
+class ActionTree(dict):
+    def __call__(self):
         app.clear_center()
-        if self.action is None:
-            widgets.ActionChoiceWidget(
-                app.center, ((k, CWFUR(v)) for k, v in self.subactions.items()),
-                horizontal=4 + (len(self.subactions) < 6)).pack()
-        else:
-            return self.action(*args, **kwargs)
+        # this can probably be done better
+        width = config.gui2.action_width
+        orphan_n = len(self) % width or float('inf')
+        orphan_nm1 = len(self) % (width - 1 or width) or float('inf')
```

Comment:
use ``width - (len % width)`` instead? Having fewer orphans sounds better. Plus, the ``ofr inf`` falls away

Reply:
balloons the second line